notion-server/src/eth_magic.rs

```diff
@@ -0,0 +1,165 @@
+use std::default;
+use std::collections::{HashMap};
+
+#[derive(Debug)]
+pub struct EthMagicManager {
+    beacon_chain: BeaconChain,
+    shard_chains: Vec<ShardChain>,
+}
+
+impl EthMagicManager {
+    pub fn new() -> Self {
+        Self {
+            beacon_chain: BeaconChain::new(),
+            shard_chains: Vec::new(),
+        }
+    }
+
+    // Creates a new execution environment on the BeaconChain and returns the
+    // index of the created execution environment
+    pub fn add_new_execution_environment(&mut self, ee: CreateEeArgs) -> u32 {
+        let execution_environment = ExecutionEnvironment {
+            wasm_code: ee.wasm_code,
+        };
+        self.beacon_chain.execution_environments.push(execution_environment);
+        (self.beacon_chain.execution_environments.len() - 1) as u32
+    }
+
+    // Returns the index of the newly added shard chain
+    // Longer-term, can accept a config here
+    pub fn add_new_shard_chain(&mut self) -> u32 {
+        let shard_chain = ShardChain::new();
+        self.shard_chains.push(shard_chain);
+        (self.shard_chains.len() - 1) as u32
+    }
+
+    // Returns the index of the newly added shard block
+    pub fn append_shard_block(&mut self, shard_index: usize, block: CreateShardBlockArgs) -> u32 {
```

Comment:
`add_new_shard_chain` returned a `u32`, but `shard_index` here is a `usize`.

Reply:
Good catch, should be `u32` here, I'll fix that.